src/app/payment/services/payment.service.ts

```diff
@@ -82,6 +83,16 @@ const paymentService = {
     return paymentsClient.getPrices();
   },
 
+  async getCoupon(): Promise<CouponAvailable> {
+    const paymentsClient = await SdkFactory.getInstance().createPaymentsClient();
```

Comment:
It could be less verbose if we do something like this: 

`return SdkFactory.getInstance().createPaymentsClient().getCoupon();`